snakemake_executor_plugin_kubernetes/__init__.py

```diff
@@ -178,28 +177,104 @@ def run_job(self, job: JobExecutorInterface):
         container.volume_mounts = [
             kubernetes.client.V1VolumeMount(name="workdir", mount_path="/workdir"),
         ]
+
+        # Volume mounts
         for pvc in self.persistent_volumes:
             container.volume_mounts.append(
                 kubernetes.client.V1VolumeMount(name=pvc.name, mount_path=str(pvc.path))
             )
 
+        # Node selector
         node_selector = {}
-        if "machine_type" in job.resources.keys():
-            # Kubernetes labels a node by its instance type using this node_label.
-            node_selector["node.kubernetes.io/instance-type"] = job.resources[
-                "machine_type"
-            ]
+        if "machine_type" in resources_dict.keys():
+            node_selector["node.kubernetes.io/instance-type"] = resources_dict["machine_type"]
+            self.logger.debug(f"Set node selector for machine type: {node_selector}")
 
+        # Add node selector for GPU if requested
+        if "gpu" in resources_dict:
+            node_selector["accelerator"] = "nvidia"
+            self.logger.debug(f"GPU requested; updated node selector: {node_selector}")
+
+        # Initialize PodSpec
         body.spec = kubernetes.client.V1PodSpec(
-            containers=[container], node_selector=node_selector
+            containers=[container],
+            node_selector=node_selector,
+            restart_policy="Never"
         )
+
+        # Add toleration for GPU nodes if GPU is requested
+        if "gpu" in resources_dict:
+            # Manufacturer logic
+            manufacturer = resources_dict.get("manufacturer", None)
+            if not manufacturer:
+                raise WorkflowError(
+                    "GPU requested but no manufacturer set. "
+                    "Use manufacturer='nvidia' or 'amd'."
```

Comment:
```suggestion
            manufacturer = resources_dict.get("gpu_manufacturer", None)
            if not manufacturer:
                raise WorkflowError(
                    "GPU requested but no manufacturer set. "
                    "Use gpu_manufacturer='nvidia' or 'amd'."
```

Reply:
Just to make this clear and distinguish it from other potential resources, I think it should be called gpu_manufacturer.